cmd/pkisetup/main.go

```diff
@@ -0,0 +1,72 @@
+/*
+   Copyright 2018 ForgeRock AS.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+  @author: Alain Pulluelo, ForgeRock (created: July 27, 2018)
+  @author: Tingyu Zeng, DELL (updated: May 21, 2019)
+  @version: 1.0.0
+*/
+
+package main
+
+import (	
+	"flag"
+	"log"
+	"os"	
+	"strconv"		
+	pki "github.com/edgexfoundry/security-secret-store/internal/pkg/pkisetup"
+)
+
+func main() {
+
+	var configFile string
+	// Handling the command flags
+	log.SetFlags(0)
+	flag.StringVar(&configFile, "config", "", "use a JSON file as configuration: /path/to/file.json")	
+	flag.Parse()
+	// Missing --config flag
+	if configFile == "" {
+		log.Println("ERROR: missing mandatory parameter: --config | -config")
+		log.Println(pki.CmdUsageMsg)
+		flag.PrintDefaults()
+		os.Exit(1)
+	}
+
+	// Read the Json config file and unmarshall content into struct type X509Config
+	log.Printf("Config file      : %s \n", configFile)
+	x509config, err := pki.ReadConfig(&configFile)
+	if err != nil {
+		pki.FatalIfErr(err, "Opening configuration file")
+	}
+
+	// Create and initialize the fs environment and global vars for the PKI materials
+	cf, err := pki.CreateEnv(&x509config)
+	if err != nil {
+		pki.FatalIfErr(err, "Environment initialization")
+	}
+
+	newCA, err := strconv.ParseBool(x509config.CreateNewRootCA)	
```

Comment:
This was implemented in the original master branch and we will need to update all the logs in the future. Given the plan to move pkisetup into edgex-go the next step, I would prefer to keep it as is if  it is working now. When we add unit test coverage next step we will be able to update these to make sure the functionality isn't broken after update.

Reply:
Note that this issue is not about having a default (as the linked issue indicates), it's more about the format of the file to begin with. If we used a JSON boolean instead of a string then we wouldn't have to do all this parsing and the code would be simpler. 

I do however recognize that changing this file format would be a breaking change in that a pkisetup json file from Edinburgh would no longer work with Fuji so that's probably not a change we can make easily, but still I think you either update that issue or file another to reflect this specific suggestion.